### PR TITLE
[RDY] Code cleanup

### PR DESCRIPTION
**Pull request by ZhilkinSerg**
# What was changed:

- added return statement after checking of null pointer;
- removed assignment of unused variables;
- removed double variable assignment;
- removed empty string initialization values;
- replaced compare with empty strings with call to `empty()` function;
- replaced assignment to empty strings with call to `clear()` function.

### CODE REVIEW

File: src/auto_pickup.cpp [hunk 1]
```diff
@@ -371,7 +371,7 @@ void auto_pickup::test_pattern(const int iTab, const int iRow)
     std::vector<std::string> vMatchingItems;
     std::string sItemName = "";
 
-    if (vRules[iTab][iRow].sRule == "") {
+    if ( !vRules[iTab][iRow].sRule.empty() ) {
```
Comment (BevapDin) [inline src/auto_pickup.cpp]: This changes the meaning, there is a new `!` at the front.

File: src/auto_pickup.h [hunk 2]
```diff
@@ -38,7 +38,7 @@ class auto_pickup
                 bool bExclude;
 
                 cRules() {
-                    this->sRule = "";
+                    this->sRule.clear();
```
Comment (BevapDin) [inline src/auto_pickup.h]: This line is completely pointless. The member `sRule` is default constructed, and that means it *is* already empty.

File: src/dependency_tree.cpp [hunk 3]
```diff
@@ -10,7 +10,7 @@ std::array<std::string, 3> error_keyvals = {{ "Missing Dependency(ies): ", "", "
 // dependency_node
 dependency_node::dependency_node(): index( -1 ), lowlink( -1 ), on_stack( false )
 {
-    key = "";
+    key.clear();
```
Comment (BevapDin) [inline src/dependency_tree.cpp]: Same here. The member `key` is already empty.

File: src/faction.cpp [hunk 4]
```diff
@@ -41,13 +41,13 @@ faction::faction()
     mapy = 0;
     size = 0;
     power = 0;
-    id = "";
-    desc = "";
+    id.clear();
+    desc.clear();
```
Comment (BevapDin) [inline src/faction.cpp]: And here. `id` and `desc` are created empty anyway, there is no need to manually clear them.

File: src/faction.cpp [hunk 5]
```diff
@@ -41,13 +41,13 @@ faction::faction()
...
 }
 
 faction::faction(std::string uid)
 {
-    name = "";
+    name.clear();
```
Comment (BevapDin) [inline src/faction.cpp]: This (and the `desc` line below) as well.

File: src/mission.cpp [hunk 6]
```diff
@@ -524,7 +524,7 @@ std::string mission::dialogue_for_topic( const std::string &in_topic ) const
 mission::mission()
 {
     type = NULL;
-    description = "";
+    description.clear();
```
Comment (BevapDin) [inline src/mission.cpp]: And here.

File: src/monster.cpp [hunk 7]
```diff
@@ -166,7 +166,7 @@ monster::monster()
     no_extra_death_drops = false;
     dead = false;
     made_footstep = false;
-    unique_name = "";
+    unique_name.clear();
```
Comment (BevapDin) [inline src/monster.cpp]: And here.

File: src/mtype.cpp [hunk 8]
```diff
@@ -14,7 +14,7 @@ mtype::mtype()
     id = mtype_id::NULL_ID();
     name = "human";
     name_plural = "humans";
-    description = "";
+    description.clear();
```
Comment (BevapDin) [inline src/mtype.cpp]: Here as well.

File: src/mutation.h [hunk 9]
```diff
@@ -45,7 +45,7 @@ struct dream {
     int strength; // The category strength required for the dream
 
     dream() {
-        category = "";
+        category.clear();
```
Comment (BevapDin) [inline src/mutation.h]: `category` does not need clearing, just as `messages` does not need clearing.

File: src/npc.cpp [hunk 10]
```diff
@@ -102,7 +102,7 @@ npc::npc()
     myclass = npc_class_id::NULL_ID();
     patience = 0;
     restock = -1;
-    companion_mission = "";
+    companion_mission.clear();
```
Comment (BevapDin) [inline src/npc.cpp]: This and the `fac_id` above don't need clearing.

File: src/options.cpp [hunk 11]
```diff
@@ -105,7 +105,7 @@ void options_manager::add_value( const std::string &lvar, const std::string &lva
 options_manager::cOpt::cOpt()
 {
     sType = "VOID";
-    sPage = "";
+    sPage.clear();
```
Comment (BevapDin) [inline src/options.cpp]: `sPage` is already created empty.

File: src/auto_pickup.cpp (hunk 1 above)
Comment (ZhilkinSerg) [reply src/auto_pickup.cpp]: Thanks! I missed this when reviewed by myself.

**Pull request merged**